ASPNetCoreMasters/Controllers/ItemsController.cs

```diff
@@ -10,32 +12,74 @@
 
 namespace ASPNetCoreMasters.Controllers
 {
+    [ApiController]
+    [Route("[controller]")]
     public class ItemsController : ControllerBase
-    {   
-        public int Get(int id)
-        {
-            //var item = new ItemService();
-            //var result = item.GetAll(id);
+    {
+        private readonly IItemService _itemService;
 
-            return id;
+        public ItemsController(IItemService itemService)
+        {
+            _itemService = itemService;
         }
 
-        public IActionResult Post([FromBody] ItemCreateBindingModel itemCreateBidingModel)
+        [HttpGet]
+        public IActionResult GetAll()
         {
-            var item = new ItemService();
-            var itemDTO = new ItemDTO();
+            var result = _itemService.GetAll();
+            return Ok(result);
+        }
 
-            if (!ModelState.IsValid)
+        [HttpGet("{itemId}")]
+        public IActionResult Get(int itemId)
+        {
+            if (String.IsNullOrEmpty(itemId.ToString()))
             {
-                return BadRequest(ModelState);
+                return BadRequest("Invalid Input");
             }
 
-            itemDTO.Text = itemCreateBidingModel.Text;
-            item.Save(itemDTO);
+            var result = _itemService.Get(itemId);
+            return Ok(result);
+        }
+
+        [HttpGet("filterBy")]
+        public IActionResult GetByFilters([FromQuery] Dictionary<string, string> filters)
+        {
+            var filter = filters.FirstOrDefault(x => x.Value == "1").Value;
```

Comment:
this line assumes that there is a dictionary with  whatever key but has a value of "1". A better approach to get values from dictionaries can be found [here ](https://www.c-sharpcorner.com/UploadFile/mahesh/dictionary-in-C-Sharp/). Since this is a key-value pair, you can look for a key named "itemId" or "text" and assign the value to your DTO.